server/index.js

```diff
@@ -29,14 +38,45 @@ app.get('/api/parks/parkCode/:parkCode', (req, res, next) => {
   fetch(`https://developer.nps.gov/api/v1/parks?parkCode=${parkCode}&limit=500&api_key=${process.env.API_KEY}`, { headers: { 'User-Agent': 'Chaim' } })
     .then(response => response.json())
     .then(data => {
-      const [ park ] = data.data;
+      const [park] = data.data;
       res.json(park);
     })
     .catch(err => {
       next(err);
     });
 });
 
+app.post('/api/parks/itineraries', (req, res, next) => {
+  const { parkCode } = req.body;
+  const userId = parseInt(req.body.userId);
+  const sql = `
+    insert into "itineraries" ("userId", "parkCode")
+    values ($1, $2)
+    returning "itineraryId"
+  `;
+  const params = [userId, parkCode];
+  db.query(sql, params)
+    .then(result => {
+      const { itineraryId } = result.rows[0];
+      const { itinerary } = req.body;
+      const newItinerary = itinerary.map(itineraryItem => {
+        return [itineraryId, itineraryItem, false];
+      });
+      const sql = format('insert into "itineraryItems" ("itineraryId", "thingToDo", "completed") VALUES %L', newItinerary);
+
+      db.query(sql)
+        .then(result => {
+
+        })
+        .catch(err => {
+          next(err);
+        });
```

Comment:
You won't need this extra `.catch()` if you include the `return` mentioned above.